src/python/deepgnn/graph_engine/snark/convert.py

```diff
@@ -113,9 +113,11 @@ def convert(self):
             num_workers=self.worker_count,
         )
 
-        for _, data in enumerate(dataset):
-            for line in data:
-                d.dispatch(line)
+        for data in dataset:
+            data_len = len(data)
+            split_len = data_len // self.thread_count + 1
```

Comment:
how would you handle a case when `data_len` = 1 and `thread_count` = 1? `split_len` will be 0, right?

Reply:
It will be 1 with the +1 at the end.